src/utils/detectionApi.ts

```diff
@@ -0,0 +1,506 @@
+  };
+  remove: string[];
+}
+
+class DetectionMapper {
+  private EVENT_TYPE: string = 'cognite_detection';
+
+  private prefix: string = 'CDF_DETECTION';
+
+  private metadataKeys: Set<string>;
+
+  private sdk: CogniteClient;
+
+  constructor(sdk: CogniteClient) {
+    this.sdk = sdk;
+    this.metadataKeys = new Set([
+      this.key('box'),
+      this.key('file_external_id'),
+      this.key('manually_verified'),
+    ]);
+  }
+
+  key(snakeCaseField: string) {
+    return `${this.prefix}_${snakeCaseField}`;
+  }
+
+  mapListResponse = (
+    eventResponse: ListResponse<CogniteEvent[]>
+  ): ListResponse<CogniteDetection[]> => {
+    return {
+      items: this.mapEventsToDetections(eventResponse.items),
+      next:
+        eventResponse.next &&
+        (() => eventResponse.next!().then(this.mapListResponse)),
+    };
+  };
+
+  mapFilter = (detectionFilter: DetectionFilter): EventFilter => {
+    return {
+      startTime: detectionFilter.startTime,
+      endTime: detectionFilter.endTime,
+      createdTime: detectionFilter.endTime,
+      lastUpdatedTime: detectionFilter.lastUpdatedTime,
+      metadata: {
+        ...detectionFilter.metadata,
+        ...(detectionFilter.fileId && {
+          [this.key('file_external_id')]: detectionFilter.fileId,
+        }),
+      },
+      assetIds: detectionFilter.assetIds,
+      rootAssetIds: detectionFilter.rootAssetIds,
+      subtype: detectionFilter.label,
+      externalIdPrefix: detectionFilter.externalIdPrefix,
+      type: this.EVENT_TYPE,
+    };
+  };
+
+  validateMetadataInput = (metadata: Metadata) => {
+    if (metadata) {
+      const invalidField: string[] = [];
+      this.metadataKeys.forEach(field => {
+        if (field in metadata) invalidField.push(field);
+      });
+      if (invalidField) {
+        throw new Error(
+          `Reserved metadata field for detections: ${invalidField}`
+        );
+      }
+    }
+  };
+
+  mapDetectionsToEvents = (items: ExternalDetection[]): ExternalEvent[] => {
+    return items.map(this.mapDetectionToEvent);
+  };
+
+  mapDetectionToEvent = (detection: ExternalDetection): ExternalEvent => {
+    if (detection.metadata) {
+      this.validateMetadataInput(detection.metadata);
+    }
+    const metadatamapped: Metadata = {};
+    if (detection.box) {
+      metadatamapped[this.key('box')] = JSON.stringify(detection.box);
+    }
+    if (detection.fileExternalId) {
+      metadatamapped[this.key('file_external_id')] = detection.fileExternalId;
+    }
+    if (detection.manuallyVerified) {
+      metadatamapped[this.key('manually_verified')] =
+        detection.manuallyVerified;
+    }
+    const metadata = this.metadataMergeInto(metadatamapped, detection.metadata);
+
+    return {
+      externalId: detection.externalId,
+      startTime: detection.startTime,
+      endTime: detection.endTime,
+      type: this.EVENT_TYPE,
+      subtype: detection.label,
+      description: detection.description,
+      metadata,
```

Comment:
Should it be this?

```suggestion
      metadata: metadata,
```

Reply:
ahh the linting would short hand this :)